osu.Game/Graphics/Containers/BeatSyncedContainer.cs

```diff
@@ -97,21 +159,16 @@ protected override void Update()
             if (timingPoint == lastTimingPoint && beatIndex == lastBeat)
                 return;
 
-            using (BeginDelayedSequence(-TimeSinceLastBeat))
-                OnNewBeat(beatIndex, timingPoint, effectPoint, track?.CurrentAmplitudes ?? ChannelAmplitudes.Empty);
+            // as this event is sometimes used for sound triggers where `BeginDelayedSequence` has no effect, avoid firing it if too far away from the beat.
+            // this can happen after a seek operation.
+            if (AllowMistimedEventFiring || Math.Abs(TimeSinceLastBeat) < MISTIMED_ALLOWANCE)
```

Comment:
While I was testing this with #12877, there's the chance that `TimeSinceLastBeat` can be equal to the length of the beat (and `TimeSinceNextBeat` becomes 0), and could be persistent across the entire gameplay, causing no metronome samples to play in the case of that PR.

I think `TimeSinceLastBeat` should add a modulo to its calculation so it becomes `0` rather than `beatLength`, not sure if that causes issues though.

Reply:
I'm confused by what this comment is trying to say (especially given that it mixes up variable names). If `TimeUntilNextBeat` can be exactly 0, then would it not be enough to modify the following guard to say `<= 0`?

https://github.com/ppy/osu/blob/6a781aedb321b205390ffa6b62b0326850ce7c51/osu.Game/Graphics/Containers/BeatSyncedContainer.cs#L92-L93

Reply:
Fixed the naming, what you've brought sounds like a correct thing as well, so that `TimeUntilNextBeat` can never be zero and `TimeSinceLastBeat` can never be `beatLength`.

Reply:
I've changed the guard. Please check to make sure you can't repro the issue any more (I couldn't make it happen but it seems feasible).

Reply:
Yeah it's dependant on the timing points I think, I reproduced it previously with https://osu.ppy.sh/beatmapsets/908896#osu/1896734.

Now it's all working properly.